docs/design/jet/025-jet-job-member-pruning.md

```diff
@@ -0,0 +1,192 @@
+with partitionKey, the submitted job could be even local.
+
+```
+- SCAN[map2, partitionKey=1]
+```
+
+##### Member pruning for Scan -> Hash Join in big cluster
+
+Prune members without any connections, so, only Member 1 and Member K left in cluster.
+
+```
+- HASH JOIN
+    - SCAN[map1, partitionKey=1]
+    - SCAN[map2, partitionKey=K]
+```
+
+![Scan + Scan -> HashJoin](https://svgshare.com/i/tfk.svg)
+
+##### Member pruning for Scan -> Transform -> Aggregate
+```
+- AGGREGATION
+  - TRANSFORM
+    - SCAN[map, partitionKey=K]
+```
+
+![Scan + Scan -> HashJoin](https://svgshare.com/i/tTL.svg)
+
+On the picture above the basic units are processors, since the actual graph looks like `Scan -> Transform -> Aggregate`.
+So, here we can eliminate whole Member 2, since according to partition key condition only Member 1 contains required data
+and final destination point `Aggregate`. Moreover, it is a good example for further processor pruning optimization.
+
+#### Solution design details
+
+To support this behaviour, we need to extend the API.
+
+We propose to change the contract of `ProcessorMetaSupplier#get` return function so that it will be
+allowed to return `null` for addresses for which it does not want to deploy processors. But this will not be enough
+to completely eliminate a member from the execution.
+
+Next, we need to determine the exact scope of the change. Two different approaches were
+considered, which differ from each other in the stage that we understand that some member
+may be pruned from job execution. The _generic_ approach tries to define the default
+behavior and applicable to DAGs from any source (SQL/Jet), whereas the _SQL-oriented_
+approach focuses on hinting `ExecutionPlanBuilder` with additional meta-information from
+JobConfig and DAG constructed by SQL optimizer.
+
+##### Generic approach
+
+The core idea here is that `ExecutionPlanBuilder` would have detailed analysis of the received DAG and possibly 
+try to optimize it before creating execution plans. The requirements to prune a member are:
+- each vertex in the DAG may work without an input;
+- DAG does not contain `distributed-broadcast` edges;
+
+This approach generifies member pruning for SQL, Pipeline API and any connector.
+However, it brings high complexity in problem analysis, long development time
+and big list of corner cases, which are discussed in the chapter below.
+
+##### SQL-oriented approach
+
+Unlike the way described above, we can think that member pruning would be beneficial mostly for small jobs.
+The overwhelming majority of such jobs are generated by SQL, and our team effort is to replace Predicate API,
+where `PartitionPredicate` is available for partitioned data querying.
+For that, we can use SQL optimization phase also for determining partitions and for all relations, we can extract information
+about required partitions to run to `JobConfig`, and then mark DAG as 'ableToPruneMembers'. This mark is an indicator
```

Comment:
`ableToPruneMembers` - seems to be outdated, there are different JobConfig paramters that are not described in TDD

Reply:
still waiting for "JobConfig paramters that are not described in TDD"